### PR TITLE
Add infraenv delete and get methods

### PR DESCRIPTION
**Pull request by nshidlin**

### CODE REVIEW

File: discovery-infra/test_infra/assisted_service_api.py [hunk 1]
```diff
@@ -127,6 +127,13 @@ def get_cluster_hosts(self, cluster_id: str) -> List[Dict[str, Any]]:
     def get_infra_env_hosts(self, infra_env_id: str) -> List[Dict[str, Any]]:
         return self.client.v2_list_hosts(infra_env_id=infra_env_id)
 
+    def get_infra_env(self, infra_env_id: str):
```
Comment (eliorerz) [inline discovery-infra/test_infra/assisted_service_api.py:130]: Add return type annotations

File: discovery-infra/test_infra/assisted_service_api.py [hunk 2]
```diff
@@ -127,6 +127,13 @@ def get_cluster_hosts(self, cluster_id: str) -> List[Dict[str, Any]]:
...
+        return self.client.get_infra_env(infra_env_id=infra_env_id)
+
+    def delete_infra_env(self, infra_env_id: str):
```
Comment (eliorerz) [inline discovery-infra/test_infra/assisted_service_api.py:133]: Add return type annotations

File: discovery-infra/test_infra/assisted_service_api.py (hunk 1 above)
Comment (nshidlin) [reply discovery-infra/test_infra/assisted_service_api.py]: done, thanks

File: discovery-infra/test_infra/assisted_service_api.py (hunk 2 above)
Comment (nshidlin) [reply discovery-infra/test_infra/assisted_service_api.py]: done, thanks

**Review by eliorerz — APPROVED** [review]
LGTM

**Review by eliorerz — APPROVED** [review]
LGTM

**Pull request merged**